packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -424,19 +424,11 @@ contract Deploy is Deployer {
     function deployL1CrossDomainMessengerProxy() public broadcast returns (address addr_) {
         console.log("Deploying proxy for L1CrossDomainMessenger");
         AddressManager addressManager = AddressManager(mustGetAddress("AddressManager"));
-        string memory contractName = "OVM_L1CrossDomainMessenger";
-        ResolvedDelegateProxy proxy = new ResolvedDelegateProxy(addressManager, contractName);
+        ResolvedDelegateProxy proxy = new ResolvedDelegateProxy(addressManager, "OVM_L1CrossDomainMessenger");
 
         save("L1CrossDomainMessengerProxy", address(proxy));
         console.log("L1CrossDomainMessengerProxy deployed at %s", address(proxy));
 
-        address contractAddr = addressManager.getAddress(contractName);
-        if (contractAddr != address(proxy)) {
-            addressManager.setAddress(contractName, address(proxy));
-        }
-
-        require(addressManager.getAddress(contractName) == address(proxy));
-
```

Comment:
@tynes can you take a look at this please? It looks to me like it was a bug, ie. we should store the implementation address, not the proxy address in the manager. 

That is handled correctly in the initialize step which calls `ProxyAdmin.upgradeAndCall()`.

Reply:
Are you saying that `setAddress` is called in `upgradeAndCall` and that we don't need to do this here?

Reply:
yes